java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogSnapshotStore.java

```diff
@@ -0,0 +1,212 @@
+    }
+
+    private Update updateLatestSnapshot(SnapshotType snapshotType, String snapshotPath, long transactionNumber, Instant updateTime) {
+        String pathKey;
+        String transactionNumberKey;
+        if (snapshotType == SnapshotType.FILES) {
+            pathKey = FILES_SNAPSHOT_PATH;
+            transactionNumberKey = FILES_TRANSACTION_NUMBER;
+        } else {
+            pathKey = PARTITIONS_SNAPSHOT_PATH;
+            transactionNumberKey = PARTITIONS_TRANSACTION_NUMBER;
+        }
+        return new Update()
+                .withTableName(latestSnapshotsTable)
+                .withKey(Map.of(TABLE_ID, createStringAttribute(sleeperTableId)))
+                .withUpdateExpression("SET " +
+                        "#Path = :path, " +
+                        "#TransactionNumber = :transaction_number, " +
+                        "#UpdateTime = :update_time")
+                .withConditionExpression("#TransactionNumber <> :transaction_number")
+                .withExpressionAttributeNames(Map.of(
+                        "#Path", pathKey,
+                        "#TransactionNumber", transactionNumberKey,
+                        "#UpdateTime", UPDATE_TIME))
+                .withExpressionAttributeValues(Map.of(
+                        ":path", createStringAttribute(snapshotPath),
+                        ":transaction_number", createNumberAttribute(transactionNumber),
+                        ":update_time", createNumberAttribute(updateTime.toEpochMilli())));
+    }
+
+    public List<TransactionLogSnapshot> getFilesSnapshots() {
+        return getSnapshots(SnapshotType.FILES)
+                .collect(Collectors.toList());
+    }
+
+    public List<TransactionLogSnapshot> getPartitionsSnapshots() {
+        return getSnapshots(SnapshotType.PARTITIONS)
+                .collect(Collectors.toList());
+    }
+
+    private Stream<TransactionLogSnapshot> getSnapshots(SnapshotType type) {
+        return streamPagedItems(dynamo, new QueryRequest()
+                .withTableName(allSnapshotsTable)
+                .withConsistentRead(true)
+                .withKeyConditionExpression("#TableIdAndType = :table_and_type")
+                .withExpressionAttributeNames(Map.of("#TableIdAndType", TABLE_ID_AND_SNAPSHOT_TYPE))
+                .withExpressionAttributeValues(new DynamoDBRecordBuilder()
+                        .string(":table_and_type", tableAndType(allSnapshotsTable, type))
+                        .build())
+                .withScanIndexForward(false)
+                .withReturnConsumedCapacity(ReturnConsumedCapacity.TOTAL))
+                .map(DynamoDBTransactionLogSnapshotStore::getSnapshotFromItem);
+    }
+
+    public Optional<TransactionLogSnapshot> getLatestFilesSnapshot() {
+        return getLatestSnapshot(SnapshotType.FILES);
+    }
+
+    public Optional<TransactionLogSnapshot> getLatestPartitionsSnapshot() {
+        return getLatestSnapshot(SnapshotType.PARTITIONS);
+    }
```

Comment:
These method signatures seem to require you to perform multiple queries to read the item. Please can they be combined?